arch/arm/src/qemu/qemu_boot.c

```diff
@@ -108,11 +106,15 @@ void arm_boot(void)
   syslog_rpmsg_init_early(g_syslog_rpmsg_buf, sizeof(g_syslog_rpmsg_buf));
 #endif
 
-#ifdef CONFIG_ARCH_ARMV7R
-  /* dont return per armv7-r/arm_head.S design */
+#ifdef CONFIG_ARM_SMP_BUSY_WAIT
+  uint32_t *address = (uint32_t *)CONFIG_ARM_SMP_BUSY_WAIT_FLAG_ADDR;
```

Comment:
let's merge into one line to avoid defining the variable in the middle of function:
```suggestion
 (uint32_t *)CONFIG_ARM_SMP_BUSY_WAIT_FLAG_ADDR = 1;
 up_flush_dcache(CONFIG_ARM_SMP_BUSY_WAIT_FLAG_ADDR,
                 CONFIG_ARM_SMP_BUSY_WAIT_FLAG_ADDR + sizeof(address));
```